### PR TITLE
[cDAC] Set RejitFlags to use a known underlying type

### PR DESCRIPTION
**Pull request by max-charlamb**
The previous implementation using an `enum` should be the size of a `uint32_t` but it isn't explicitly defined. To make the size of `RejitFlags` more clear, changing it to an `enum class` with an explicit `uint32_t` backing.

Will be used #109936 by the cDAC `GetMethodDescData` to fetch `ReJIT` information.

### CODE REVIEW

**Review by davmason — APPROVED** [review]
LGTM modulo the comment about usage.

File: src/coreclr/vm/codeversion.h [hunk 1]
```diff
@@ -184,7 +184,7 @@ class ILCodeVersion
     HRESULT SetActiveNativeCodeVersion(NativeCodeVersion activeNativeCodeVersion);
 #endif //DACCESS_COMPILE
 
-    enum RejitFlags
+    enum class RejitFlags : uint32_t
```
[the remark this answers is not in the document]
Comment (max-charlamb) [reply src/coreclr/vm/codeversion.h:187]: I don't think this is possible to do for `enum class`'s in C++14. It looks like `using enum` is only available in C++20 and higher. 

I see some approaches online manually dumping each enum value, but that feels worse than referencing them directly. https://stackoverflow.com/a/48932002

[the remark this answers is not in the document]
Comment (davidwrighton) [reply src/coreclr/vm/codeversion.h:187]: It is a bit unfortunate. Sometimes in the past I've simply elevated the enum to no longer be contained within the class when I do this conversion. Then I can just use RejitFlags::BLAH wherever it needs to be used. You still have to touch all of the places where the enum is touched, but the major reason to put an enum within `ILCodeVersion` was probably to give its members a namespace to live in, and enum class does that all by itself.

**Review by davidwrighton — APPROVED** [review]
I've made another comment about how I might have pulled the enum out of the `ILCodeVersion` class to avoid the `ILCodeVersion::RejitState` prefix on everything, but it is not critical to me.

**Pull request merged**